async_examples/event_test.ml

```diff
@@ -76,7 +76,7 @@ let watch_events rpc session_id =
     let open Event_types in
     Event.from ~rpc ~session_id ~classes:["*"] ~token:"" ~timeout:0. >>= fun rpc ->
     let e = event_from_of_rpc rpc in
-    if e.events = [] then error "Empty list of events";
+    if List.length e.events = 0 then error "Empty list of events";
```

Comment:
Why is this needed or better?

Reply:
`=` is an alias to the monomorphic `Core.Int.(=)` so it cannot be used here to compare lists.

Alternatives are:
1. Use a prefixed equal (something like `List.equal Event.equal e.events []`), but I don't think Event.equal is of easy access.
2. Remove the `Open.Core`. This breaks quite a few lines in the file so I avoided it.
3. Use length to be able to use integer comparison as is. I chose this because it seemed like the less disruptive choice.